### ISSUE TITLE
How is a remote account able to see attached media for followers-only?

### ISSUE THREAD

**Issue by gbraad** [friendica/friendica#12554]
I created a message addressed to 'Followers only', and attached media using the browser option. The message is shown as follows on the receiver (a Mastodon server):

![image](https://user-images.githubusercontent.com/1894/209764358-d74412bf-8358-4261-8d66-f7b950462de2.png)

The media is shown as "Not Available" and clicking on it also reveals that there is no such media?

https://social.gbraad.nl/photo/156234278563abd62f0c163082392668-0.jpg

![image](https://user-images.githubusercontent.com/1894/209764907-60fb33f9-fdc7-4a0b-aa71-8b65227805a4.png)

When logged in the media is available. This means the receiver 'technically' needs to have an account on the originating server before he is able to open the media?

---

This media is marked as 'Followers only', so this needs to be made public before it can show up on remote servers/accounts:

![image](https://user-images.githubusercontent.com/1894/209764768-e7e87f26-4108-4714-aa82-b451e6c96a56.png)

Note: I changed this to 'Public' as otherwise this would not work.

**Comment by gbraad** [comment]
This media is marked as 'Followers only', so this needs to be made public before it can show up on remote servers/accounts:

![image](https://user-images.githubusercontent.com/1894/209764768-e7e87f26-4108-4714-aa82-b451e6c96a56.png)

**Comment by tobiasd** [comment]
You can make all posted pictures accessible in your account settings, the "Security and Privacy" section and there you find an option called "Make all posted pictures accessible".

**Comment by gbraad** [comment]
It might be helpful to display a message to the user when remote followers are involved that the image might not be visible to them (unless permission given either way).

**Comment by MrPetovan** [comment]
The way Friendica remote account can view private pictures is through magic authentication. Basically, they’re able to authenticate as visitors on your node after a cryptographic challenge is solved by their server.

Of course this doesn’t work with any other system 🤷‍♂️

**Comment by gbraad** [comment]
> Of course this doesn’t work with any other system

Is this documented as a specification ?

**Comment by MrPetovan** [comment]
> Is this documented?

Ha ha ha hahaha ha haha ha, no.

**Comment by gbraad** [comment]
> no.

I figured. Is this what the `"magic-public-key"` is used for?
Seems I might have a little bit more work on hands.
A draft spec can already help with review/scrutiny.

**Comment by MrPetovan** [comment]
It would but at the same time Friendica is a minor platform so it’s unlikely any other would implement magic auth.

Besides, the Fediverse trend is to have a copy of everything remote on your local server, so you don’t have to navigate to remote servers; so while it’s been working, it’s unlikely to ever get any traction. Even we are moving away from it.

**Comment by gbraad** [comment]
Is this related to Hubzilla's magic-auth?

**Comment by MrPetovan** [comment]
It is most likely the same concept, not sure if the implementation matches.

**Comment by tobiasd** [comment]
> Is this related to Hubzilla's magic-auth?

Yes. Magic-Auth comes from a time before Hubzilla was forked off from Friendica. But Hubzilla extended the functionality and spiced it with the nomadic identities.

**Comment by annando** [comment]
Isn't that the stuff that @rabuzarus ported from Hubzilla to Friendica?

**Comment by tobiasd** [comment]
He ported some of the magic enhancements to the remote auth back to Friendica yes.

**Comment by gbraad** [comment]
Is this compatible with Hubzilla in that case?

**Comment by annando** [comment]
I don't know. But also for stuff like these pictures we moved to signed http requests.

**Comment by gbraad** [comment]
@annando how does this work?

**Comment by annando** [comment]
We are fetching pictures with the signature of the requested user. The target system then can check the signature and can verify that it belongs to an authorized user and will serve the picture.

**Comment by gbraad** [comment]
where in the codebase can this be found?
